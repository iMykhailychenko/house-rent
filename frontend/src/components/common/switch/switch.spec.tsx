import { fireEvent, render } from '@testing-library/react';

import Switch from './switch';

describe('Switch component', () => {
    describe('Switch small', () => {
        it('matches snapshot', () => {
            const { asFragment } = render(<Switch size="sm" className="test" value={false} width={10} />);
            expect(asFragment()).toMatchSnapshot();
        });

        it('toggle small', () => {
            const handleChange = jest.fn();
            const { getByRole } = render(<Switch size="sm" value={false} onChange={handleChange} />);
            fireEvent.click(getByRole('button'));
            expect(handleChange).toHaveBeenCalledWith(true);
            expect(handleChange).toHaveBeenCalledTimes(1);
        });
    });

    describe('Switch large', () => {
        it('matches snapshot', () => {
            const { asFragment } = render(
                <Switch size="lg" className="test" value={false} labels={['test-left', 'test-right']} width={10} />,
            );
            expect(asFragment()).toMatchSnapshot();
        });

        it('toggle large', () => {
            const handleChange = jest.fn();
            const { getByRole } = render(
                <Switch size="lg" value={false} onChange={handleChange} labels={['test-left', 'test-right']} />,
            );
            fireEvent.click(getByRole('button'));
            expect(handleChange).toHaveBeenCalledWith(true);
            expect(handleChange).toHaveBeenCalledTimes(1);
        });
    });
});
