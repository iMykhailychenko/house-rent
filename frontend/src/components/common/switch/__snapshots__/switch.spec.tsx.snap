// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component Switch large matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="root right test"
    type="button"
  >
    <span
      class=""
      style="min-width: 10rem;"
    >
      test-left
    </span>
    <span
      class="active"
      style="min-width: 10rem;"
    >
      test-right
    </span>
  </button>
</DocumentFragment>
`;

exports[`Switch component Switch small matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="root right test"
    type="button"
  >
    <span
      class="indicator"
    />
  </button>
</DocumentFragment>
`;
